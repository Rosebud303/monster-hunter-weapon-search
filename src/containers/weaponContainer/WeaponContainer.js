import React, { Component } from 'react';
import { connect } from 'react-redux';
import { Route } from 'react-router-dom'
// import { getCurrentWeapon } from '../../actions'
import WeaponDetails from '../../components/weaponDetails/WeaponDetails'
import Weapon from '../../components/weapon/Weapon'
import './WeaponContainer.scss'


export class WeaponContainer extends Component {

  render() {
    let displayCards = this.props.weapons.map(weapon => {
      return <Weapon {...weapon} key={weapon.id}/>
    })
    return (
      <div className='weapon-container'>
        {displayCards}
      </div>
    )
  }
}

export const mapStateToProps = (state) => ({
  weapons: state.weapons,
  currentWeapon: state.currentWeapon
})

export default connect(mapStateToProps)(WeaponContainer)