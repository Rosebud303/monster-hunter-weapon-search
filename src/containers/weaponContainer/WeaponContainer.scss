.weapon-container {
  display: flex;
  flex-wrap: wrap;
  justify-content: space-around;
  max-height: 58%;
  overflow: scroll;
  box-shadow: 0px 1px 20px 5px;
  background-color: rgba(0, 0, 0, 0.6);
  border: solid 4px #231f20c2;
  padding: 2em;
  width: 80%;
  margin: auto;
}