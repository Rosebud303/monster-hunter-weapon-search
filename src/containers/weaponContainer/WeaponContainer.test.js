import React from 'react';
import { shallow } from 'enzyme';
import { WeaponContainer, mapStateToProps } from './WeaponContainer';

let mockWeps = [{
  craftable: true,
  damageType: "sever",
  id: 1,
  maxDMG: 384,
  minDMG: 80,
  name: "Buster Sword 1",
  picture: "https://assets.mhw-db.com/weapons/great-sword/83f7ab6e7e5669ec416d772049b8b054e2624c2e.c7bad811e203c9bb55626e414659a4f7.png",
  rarity: 1,
  type: "great-sword",
}, {
  craftable: true,
  damageType: "sever",
  id: 2,
  maxDMG: 384,
  minDMG: 80,
  name: "Buster Sword 2",
  picture: "https://assets.mhw-db.com/weapons/great-sword/83f7ab6e7e5669ec416d772049b8b054e2624c2e.c7bad811e203c9bb55626e414659a4f7.png",
  rarity: 1,
  type: "great-sword",
}]

describe('WeaponContainer', () => {
  
  describe('Component', () => {
    let wrapper;  

    beforeEach(() => {
      wrapper = shallow (
        <WeaponContainer weapons={mockWeps}/>
      )
    });

    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot()
    });
  });

  describe('mapStateToProps', () => {
    const mockState = {
      isLoading: false,
      weapons: mockWeps
    }
    const expected = {
      weapons: mockWeps
    }
    const result = mapStateToProps(mockState)
    expect(result).toEqual(expected)
  })
  
});