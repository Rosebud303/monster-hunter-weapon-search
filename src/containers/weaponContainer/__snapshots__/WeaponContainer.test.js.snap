// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeaponContainer Component should match the snapshot 1`] = `
<div
  className="weapon-container"
>
  <Weapon
    craftable={true}
    damageType="sever"
    id={1}
    key="1"
    maxDMG={384}
    minDMG={80}
    name="Buster Sword 1"
    picture="https://assets.mhw-db.com/weapons/great-sword/83f7ab6e7e5669ec416d772049b8b054e2624c2e.c7bad811e203c9bb55626e414659a4f7.png"
    rarity={1}
    type="great-sword"
  />
  <Weapon
    craftable={true}
    damageType="sever"
    id={2}
    key="2"
    maxDMG={384}
    minDMG={80}
    name="Buster Sword 2"
    picture="https://assets.mhw-db.com/weapons/great-sword/83f7ab6e7e5669ec416d772049b8b054e2624c2e.c7bad811e203c9bb55626e414659a4f7.png"
    rarity={1}
    type="great-sword"
  />
</div>
`;
